### PR TITLE
ROX-19190: refactoring

### PR DESCRIPTION
**Pull request by 0x656b694d**
* fixes to the start-acs action
* refactoring of the start-acs.sh script
* set telemetry key to target the dev instance

### CODE REVIEW

**Comment by 0x656b694d** [comment]
Current dependencies on/for this PR:

* main
  * **PR #23** 👈

**Review by vikin91 — COMMENTED** [review]
Thanks for picking it! Have you already had a chance to test it?

**Comment by 0x656b694d** [comment]
> Thanks for picking it! Have you already had a chance to test it?

Not tested yet. I'll ask for support :)

**Comment by 0x656b694d** [comment]
Tested here: https://github.com/stackrox/test-gh-actions/actions/runs/6009249439/job/16298905676

It failed kube-burner, as the test test-gh-actions branch diverged from main, and is unrelated to the change in this PR.

File: release/start-acs/start-acs.sh [hunk 1]
```diff
@@ -28,28 +28,81 @@ pushd "$STACKROX_DIR"
...
+                { "name": "ROX_FAKE_KUBERNETES_WORKLOAD", "value": "long-running" },
+                { "name": "ROX_FAKE_WORKLOAD_STORAGE", "value": "/var/cache/stackrox/pebble.db" }
+            ],
+            "resources": {
+                "requests": { "memory": "3Gi", "cpu": "2" },
+                "limits": { "memory": "12Gi", "cpu": "4" }
+            }
+        }
+    ] } }
+} }
```
Comment (vikin91) [inline release/start-acs/start-acs.sh:56]: Could we pass it through json-pretty to unify the indents?

File: release/start-acs/start-acs.sh [hunk 2]
```diff
@@ -28,28 +28,81 @@ pushd "$STACKROX_DIR"
...
+gh_log notice "Forwarding central port..."
 kubectl -n stackrox port-forward deploy/central 8000:8443 > /dev/null 2>&1 &
 sleep 20
 
+gh_log notice "Deploying sensor..."
 "${STACKROX_DIR}"/deploy/k8s/sensor.sh
 
-kubectl -n stackrox set env deploy/sensor MUTEX_WATCHDOG_TIMEOUT_SECS=0 ROX_FAKE_KUBERNETES_WORKLOAD=long-running ROX_FAKE_WORKLOAD_STORAGE=/var/cache/stackrox/pebble.db
-kubectl -n stackrox patch deploy/sensor -p '{"spec":{"template":{"spec":{"containers":[{"name":"sensor","resources":{"requests":{"memory":"3Gi","cpu":"2"},"limits":{"memory":"12Gi","cpu":"4"}}}]}}}}'
+PATCH=$(cat <<EOPATCH
```
Comment (vikin91) [inline release/start-acs/start-acs.sh:41]: Quotes `"` are missing around `$()`.

Can we use heredoc instead of cat?

Comment (vikin91) [reply release/start-acs/start-acs.sh:41]: Or even better: this patch is static and does not contain any variables - can we save it as a separate json file?

File: release/start-acs/start-acs.sh [hunk 3]
```diff
@@ -28,28 +28,81 @@ pushd "$STACKROX_DIR"
...
+            "env": [
+                { "name": "MUTEX_WATCHDOG_TIMEOUT_SECS", "value": "0" }
+            ],
+            "resources": {
+                "requests": { "memory": "3Gi", "cpu": "2" },
+                "limits": { "memory": "12Gi", "cpu": "4" }
+            }
+        }
+    ] } }
+} }
```
Comment (vikin91) [inline release/start-acs/start-acs.sh:75]: This patch is a subset of the sensor's one - there is a potential to share the common parts. Could we have it as a separate file as well?

**Review by vikin91 — COMMENTED** [review]
Looks good, but I think there is potential for refactoring the patches as well

File: release/start-acs/start-acs.sh (hunk 3 above)
Comment (0x656b694d) [reply release/start-acs/start-acs.sh]: That would require a messy json operations, combining several pieces of json, which would be less clear than having two similar json files. I think.

File: release/start-acs/start-acs.sh (hunk 2 above)
Comment (0x656b694d) [reply release/start-acs/start-acs.sh]: Done.

File: release/start-acs/start-acs.sh (hunk 1 above)
Comment (0x656b694d) [reply release/start-acs/start-acs.sh]: Done. Extracted to a separate file.

**Review by vikin91 — APPROVED** [review]
Thanks for improving!

**Comment by 0x656b694d** [comment]
Another test run after latest changes: https://github.com/stackrox/test-gh-actions/actions/runs/6094942424/job/16538247997

**Pull request merged**